### PR TITLE
azure-events: Updated agent to use ocf.py, Python 2/3 compatibility

### PR DESCRIPTION
**Pull request by krig**
This pull request is an updated version of the agent proposed in #1161, see discussion there for background on the changes made.

The biggest complication was the use of `enum`, which is only available as a third-party module in Python 2.7 or Python 3.3 and lower (called `enum34`). In order to avoid adding `enum34` as a dependency, the code got a bit uglier. I am going to revisit that later, I think I can do it in a more maintainable way by making the states into objects rather than integers but still avoiding the `enum` module.

I have done some basic testing on Azure and so far it seems to work as intended.

### CODE REVIEW

**Comment by krig** [comment]
@tniek Unfortunately Github wouldn't let me create a pull request against your branch, so I ended up creating a new pull request. Comments / suggestions / opinions as well as help testing are more than welcome. :)

**Comment by tniek** [comment]
@krig Thanks a lot for making these changes! I've tested the agent in an actual HA pair deployment and functionality-wise, everything still seems to work as expected.

**Comment by krig** [comment]
@tniek Excellent, thank you!

@oalbrigt Would you be OK with merging?

File: heartbeat/azure-events.in [hunk 1]
```diff
@@ -0,0 +1,820 @@
...
+	def _sendMetadataRequest(endpoint, postData=None):
+		"""
+		Send a request to Azure's Azure Metadata Service API
+		"""
+		url = "%s/%s?api-version=%s" % (azHelper.metadata_host, endpoint, azHelper.api_version)
+		ocf.logger.debug("_sendMetadataRequest: begin; endpoint = %s, postData = %s" % (endpoint, postData))
+		ocf.logger.debug("_sendMetadataRequest: url = %s" % url)
+
+		req = urllib2.Request(url, postData)
+		req.add_header("Metadata", "true")
```
Comment (harijayms) [inline heartbeat/azure-events.in]: Could we please add a useragent string here with value of pacemaker. This would allow us to know the calls source.

Comment (harijayms) [reply heartbeat/azure-events.in]: @tniek  can you push updates in this PR for the same ?

File: heartbeat/azure-events.in [hunk 2]
```diff
@@ -6,56 +6,39 @@
...
-OCF_ERR_CONFIGURED = 6
-OCF_NOT_RUNNING = 7
-
 attr_globalPullState = "azure-events_globalPullState"
 attr_lastDocVersion  = "azure-events_lastDocVersion"
 attr_curNodeState = "azure-events_curNodeState"
 attr_pendingEventIDs = "azure-events_pendingEventIDs"
 
-default_loglevel = logging.INFO
+default_loglevel = ocf.logging.INFO
```
Comment (oalbrigt) [inline heartbeat/azure-events.in]: I guess this should be logger.

**Comment by oalbrigt** [comment]
Looks good in general. Just check the logger-comment and cherry-pick the useragent @harijayms mentioned above.

File: heartbeat/azure-events.in (hunk 1 above)
Comment (diegoakechi) [reply heartbeat/azure-events.in]: @krig copying from the old PR, @harijayms is talking about this HTTP header:

`req.add_header('User-agent', 'Pacemaker-%osversion%')`

**Comment by krig** [comment]
Alright, will do.

File: heartbeat/azure-events.in (hunk 1 above)
Comment (krig) [reply heartbeat/azure-events.in]: I'll add this, thank you for the reminder!

File: heartbeat/azure-events.in (hunk 2 above)
Comment (krig) [reply heartbeat/azure-events.in]: No, I actually changed that to logger but apparently the INFO/DEBUG/etc. constants only exist on the module itself.

File: heartbeat/azure-events.in (hunk 1 above)
Comment (krig) [reply heartbeat/azure-events.in]: Done, the User-Agent header will have the format `Pacemaker-ResourceAgent/<VERSION>`.

File: heartbeat/azure-events.in (hunk 2 above)
Comment (oalbrigt) [reply heartbeat/azure-events.in]: Oh. Good to check anyways :)

**Comment by krig** [comment]
I've updated the PR with a commit setting the user agent, but I want to dig a bit and see if I can include a bit more detail.

File: heartbeat/azure-events.in [hunk 3]
```diff
@@ -24,7 +24,31 @@ import ocf
 
 ##############################################################################
 
+def detect_distro():
```
Comment (oalbrigt) [inline heartbeat/azure-events.in]: I guess we should move this to ocf. Maybe just ocf.distro()?

Comment (krig) [reply heartbeat/azure-events.in]: Yeah, that would make sense. I can add it in a separate PR.

Comment (oalbrigt) [reply heartbeat/azure-events.in]: Sure. Then you're free to merge it if you're done.

Comment (krig) [reply heartbeat/azure-events.in]: Perfect, thanks.

**Comment by krig** [comment]
Thanks everyone for your comments!

**Pull request merged**